src/lib/quoter.js

```diff
@@ -39,77 +125,44 @@ class Quoter {
     if (request.destinationAmount === '0') {
       throw new InvalidAmountSpecifiedError('destinationAmount must be positive')
     }
-    return yield this._quoteByAmount(request)
+    // Use the shifted curve because the `amountReverse()` is rounded down,
+    // and we don't want to lose money.
+    const liquidityQuote = yield this._quoteLiquidity(Object.assign({_shiftCurve: true}, request))
+    if (!liquidityQuote) return null
+    return Object.assign({
+      sourceAmount: liquidityQuote.liquidityCurve.amountReverse(request.destinationAmount).toString()
+    }, liquidityQuote)
   }
 
-  * _quoteByAmount (request) {
-    const destinationHoldDuration = request.destinationHoldDuration
-    const hop = this._findBestHopForAmount(request)
-    if (!hop) return null
-
-    // note the confusion between `next`, `destination`, and `final`;
-    // `finalAmount` will be renamed to `destinationAmount` later in
-    // the hopToQuote function.
-    if (hop.sourceAmount && hop.finalAmount) {
-      return hopToQuote(hop, destinationHoldDuration)
-    }
-
-    // Otherwise, ask a connector closer to the destination.
-
-    let headHop
-    const intermediateConnector = hop.destinationCreditAccount
-    // Quote by source amount
-    if (request.sourceAmount) {
-      headHop = this.tables.findBestHopForSourceAmount(
-        hop.sourceLedger, intermediateConnector, request.sourceAmount)
-    }
-    const tailQuote = yield ILQP.quoteByConnector({
-      plugin: this.ledgers.getPlugin(hop.destinationLedger),
-      connector: intermediateConnector,
-      quoteQuery: {
-        sourceAmount: request.destinationAmount === undefined ? headHop.destinationAmount : null,
-        destinationAmount: request.sourceAmount === undefined ? hop.finalAmount : null,
-        destinationAccount: request.destinationAccount,
-        destinationHoldDuration
-      }
+  /**
+   * @param {IlpAddress} sourceLedger
+   * @param {IlpAddress} destination
+   * @param {Amount} sourceAmount
+   * @returns {Object}
+   */
+  * findBestPathForSourceAmount (sourceLedger, destination, sourceAmount) {
+    const quote = yield this.quoteBySourceAmount({
```

Comment:
This looks like paths it is going to trigger a quote request, which is not necessarily the case.
Maybe add a code comment explaining that this function ends up calling https://github.com/interledgerjs/ilp-routing/blob/master/src/lib/routing-table.js#L38 so the routing table is checked first, and remote quoting is only used if the route's curve is expired or missing.